pkgs/misc/uboot/default.nix

```diff
@@ -366,6 +391,103 @@ in {
+        sha256 = "sha256-bfceynwI4d4bLJvkg8OwUY2N65C02Yl3KFmpE7N6kAk=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/enable-DT-overlays-support.patch";
+        sha256 = "sha256-xm4/yZlCNDyi9QfdUnpOqjsEeIFFc+BDkYy2qnblD74=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-add-xtx-spi-nor-chips.patch";
+        sha256 = "sha256-fHaTwUXDO8CcX6RRyMgEyZ7CZX0LzEha/IKulGRQbzw=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-dwc-otg-usb-fix.patch";
+        sha256 = "sha256-e1MsmAxkSXB2qHmSkD7DSv56VhcEJmIN1c2XuLZPMKM=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-set-eth1addr.patch";
+        sha256 = "sha256-u/kiWf0vz3Yw+rEgAO+BPw/PkKom3SM2F9jOS/6/Nu0=";
+      })
+        (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-support-recovery-button.patch";
+        sha256 = "sha256-uP1eLOALtxS7Ue9heVAkYFDwvW0lTZdQfhVioh5oBNc=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-support-rmii-integrated-phy.patch";
+        sha256 = "sha256-y4FaQ1OQGVvkD8Sot0jNJMuGO6p9o2u03JiL5MKihm4=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/rk3328-efuse-driver.patch";
+        sha256 = "sha256-ly2dJb1BRXuITfNYCiqqIxqErWQnrdonhpaLwmeKAm4=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/rk3328-resets-for-mmc-controllers.patch";
+        sha256 = "sha256-abd2V0D2Cm5hw+JiWa80c7kmhcV8G6wbyyxw/ZAWu+0=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/rk3328-sdmmc_ext-node.patch";
+        sha256 = "sha256-EHS/aBik4V4pbEU9GHhCXCBgdoExbVvUzB01KDx4zM8=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/sdmmc-force-fifo-mode-in-spl.patch";
+        sha256 = "sha256-+oy6kG/wAvd89M+ZS6a2xlKWzzdHV6UCtd1SaTmLJaI=";
+      })
+    ];
```

Comment:
We want to support mainline U-Boot in Nixpkgs, having this many patches is problematic.

AFAICT this is basically adding everything needed for support in U-Boot, right?

Sadly, it looks like whatever work armbian is using:

 - Some changes are missing any origin information
 - They have apparently not sent this for upstreaming [(1)](https://patchwork.ozlabs.org/project/uboot/list/?series=&submitter=&state=*&q=orangepi-r1-plus&archive=both&delegate=)[(2)](https://patchwork.ozlabs.org/project/uboot/list/?series=&submitter=&state=*&q=rk3328-orangepi-r1-plus-lts&archive=both&delegate=)